karbon/karbon_api.go

```diff
@@ -22,6 +24,69 @@ type Client struct {
 	Meta            MetaService
 }
 
+type ClientService interface {
+	CreateK8sRegistration(createRequest *K8sCreateClusterRegistrationRequest) (*K8sCreateClusterRegistrationResponse, error)
+	DeleteK8sRegistration() (*K8sClusterRegistrationDeleteResponse, error)
+	GetK8sRegistration() (*K8sClusterRegistration, error)
+	GetK8sRegistrationList() (*K8sClusterRegistrationList, error)
+}
+
+func (op Client) CreateK8sRegistration(createRequest *K8sCreateClusterRegistrationRequest) (*K8sCreateClusterRegistrationResponse, error) {
+	ctx := context.TODO()
+
+	path := "/v1-alpha.1/k8s/cluster-registrations/"
+	req, err := op.httpClient.NewRequest(http.MethodPost, path, createRequest)
+	karbonClusterActionResponse := new(K8sCreateClusterRegistrationResponse)
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Consider moving this to line 39. Errors should be checked right after a function call where they get returned.